mm2src/coins/eth.rs

```diff
@@ -2478,13 +2480,36 @@ async fn sign_and_send_transaction_with_metamask(
         Action::Call(to) => Some(to),
     };
 
-    let gas_price = try_tx_s!(coin.get_gas_price().compat().await);
+    let pay_for_gas_option = try_tx_s!(
+        coin.get_swap_pay_for_gas_option(coin.get_swap_transaction_fee_policy())
+            .compat()
+            .await
+    );
 
     let tx_to_send = TransactionRequest {
         from: coin.my_address,
         to,
         gas: Some(gas),
-        gas_price: Some(gas_price),
+        gas_price: if let PayForGasOption::Legacy(LegacyGasPrice { gas_price }) = pay_for_gas_option {
+            Some(gas_price)
+        } else {
+            None
+        },
+        max_priority_fee_per_gas: if let PayForGasOption::Eip1559(Eip1559FeePerGas {
+            max_priority_fee_per_gas,
+            ..
+        }) = pay_for_gas_option
+        {
+            Some(max_priority_fee_per_gas)
+        } else {
+            None
+        },
+        max_fee_per_gas: if let PayForGasOption::Eip1559(Eip1559FeePerGas { max_fee_per_gas, .. }) = pay_for_gas_option
+        {
+            Some(max_fee_per_gas)
+        } else {
+            None
+        },
```

Comment:
same, please move fields calculation before `let tx_to_send = TransactionReques` to improve readability

Reply:
done

Reply:
here please use match pattern. 

PS: I there is a code duplication in three places. Please create priv function to create  gas price values and use it.

Here two. pllus one from this comment

https://github.com/KomodoPlatform/komodo-defi-framework/blob/66f8ff76558e8330d08c6694d9e4efc6dbc3b7a3/mm2src/coins/eth.rs#L768-L782

https://github.com/KomodoPlatform/komodo-defi-framework/blob/66f8ff76558e8330d08c6694d9e4efc6dbc3b7a3/mm2src/coins/eth.rs#L2506-L2520

```rust
            let gas_price = match pay_for_gas_option {
                PayForGasOption::Legacy(ref legacy_gas_price) => Some(legacy_gas_price.gas_price),
                _ => None,
            };

            let (max_priority_fee_per_gas, max_fee_per_gas) = match pay_for_gas_option {
                PayForGasOption::Eip1559(Eip1559FeePerGas {
                    max_priority_fee_per_gas,
                    max_fee_per_gas,
                }) => (Some(max_priority_fee_per_gas), Some(max_fee_per_gas)),
                _ => (None, None),
            };
```

Reply:
fixed in [0ec27e9](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2051/commits/0ec27e914bd8d3eb0dd40d1ff9500dad459ff742)